source/isaaclab/isaaclab/sensors/frame_transformer/frame_transformer.py

```diff
@@ -423,38 +430,33 @@ def _set_debug_vis_impl(self, debug_vis: bool):
             if not hasattr(self, "frame_visualizer"):
                 self.frame_visualizer = VisualizationMarkers(self.cfg.visualizer_cfg)
 
-                try:
-                    # isaacsim.util is not available in headless mode
-                    import isaacsim.util.debug_draw._debug_draw as isaac_debug_draw
-
-                    self.debug_draw = isaac_debug_draw.acquire_debug_draw_interface()
-                except ImportError:
-                    omni.log.info("isaacsim.util.debug_draw module not found. Debug visualization will be limited.")
-
             # set their visibility to true
             self.frame_visualizer.set_visibility(True)
         else:
             if hasattr(self, "frame_visualizer"):
                 self.frame_visualizer.set_visibility(False)
-                # clear the lines
-                if hasattr(self, "debug_draw"):
-                    self.debug_draw.clear_lines()
 
     def _debug_vis_callback(self, event):
-        # Update the visualized markers
-        all_pos = torch.cat([self._data.source_pos_w, self._data.target_pos_w.view(-1, 3)], dim=0)
-        all_quat = torch.cat([self._data.source_quat_w, self._data.target_quat_w.view(-1, 4)], dim=0)
-        self.frame_visualizer.visualize(all_pos, all_quat)
-
-        if hasattr(self, "debug_draw"):
-            # Draw lines connecting the source frame to the target frames
-            self.debug_draw.clear_lines()
-            # make the lines color yellow
-            source_pos = self._data.source_pos_w.cpu().tolist()
-            colors = [[1, 1, 0, 1]] * self._num_envs
-            for frame_index in range(len(self._target_frame_names)):
-                target_pos = self._data.target_pos_w[:, frame_index].cpu().tolist()
-                self.debug_draw.draw_lines(source_pos, target_pos, colors, [1.5] * self._num_envs)
+        # Get the all frames pose
+        frames_pos = torch.cat([self._data.source_pos_w, self._data.target_pos_w.view(-1, 3)], dim=0)
+        frames_quat = torch.cat([self._data.source_quat_w, self._data.target_quat_w.view(-1, 4)], dim=0)
+
+        # Get the all connecting lines between frames pose
+        lines_pos, lines_quat, lines_scale = self._get_connecting_lines(
+            source_pos=self._data.source_pos_w.repeat_interleave(self._data.target_pos_w.size(1), dim=0),
+            target_pos=self._data.target_pos_w.view(-1, 3),
+        )
+
+        # Update the frame and the connecting line visualizer
+        self.frame_visualizer.visualize(
+            translations=torch.cat((frames_pos, lines_pos), dim=0),
+            orientations=torch.cat((frames_quat, lines_quat), dim=0),
+            scales=torch.cat((torch.ones(frames_pos.size(0), 3, device=frames_pos.device), lines_scale), dim=0),
```

Comment:
NIT: Some parts could be optimized by initializing the constant tensors once. However, since debug visualization only ever happens for small envs at play time, might not be a huge consideration.

Reply:
@Mayankm96 Moved tensor initialization to `_set_debug_vis_impl` so that it is initialized once.